### PR TITLE
[Testing] Resolve all promises

### LINKED ISSUE #3828: [Testing] Fix dangling promises in unit tests
## Overview
Some promises are not being resolved

## Expected Behavior
All promises should be resolved

## Steps to Reproduce
1. npm run test

### PR DESCRIPTION
**Pull request by davetsay**
## Overview
Fixes #3828 , #3934

- fix unresolved promises

## Testing Instructions 
See #3828 

## Author Checklist
| | |
| --- | :---: |
| Changes address original issue? | Y |
| Unit tests included and/or updated with changes? | Um... yeah |
| Command line build passes? | Y |
| Changes have been smoke-tested? | Y |
| Testing instructions included? | Y |

### CODE REVIEW

**Comment by davetsay** [comment]
Dunno if changing `resetApplicationState` to `resetApplicationStatePromise` is too weird. But, thats what reviews are for. There were quite a few resetApplicationStates that were not returned or awaited on.

**Review by shefalijoshi — DISMISSED** [review]
LGTM

File: src/plugins/LADTable/pluginSpec.js [hunk 1]
```diff
@@ -398,15 +389,13 @@ describe("The LAD Table Set", () => {
 
             await Promise.all([telemetryRequestPromise, ladObjectPromise, anotherLadObjectPromise]);
             await Vue.nextTick();
-
-            return;
         });
 
-        it("should show one row per lad table object in the composition", () => {
+        // TODO this should be fixed
+        xit("should show one row per lad table object in the composition", () => {
```
Comment (akhenry) [inline src/plugins/LADTable/pluginSpec.js:395]: Any way we can fix this?

**Review by akhenry — CHANGES_REQUESTED** [review]
This looks great, it's a lot of work!

I am unsure about naming a function `resetApplicationStatePromise()`. I totally understand the motivation behind renaming it so that it's more obvious that it returns a promise. As a developer though I would expect something named `resetApplicationStatePromise` to be a promise object.

I *think* that just fixing the existing usage of this function will make the problem go away? People copy existing test specs to create new ones, so they will copy the correct usage.

Also, if we could fix that x-ed out test that would be awesome.

File: src/plugins/LADTable/pluginSpec.js (hunk 1 above)
Comment (davetsay) [reply src/plugins/LADTable/pluginSpec.js]: I didn't think so, no. But its fixed.

**Comment by davetsay** [comment]
> This looks great, it's a lot of work!
> 
> I am unsure about naming a function `resetApplicationStatePromise()`. I totally understand the motivation behind renaming it so that it's more obvious that it returns a promise. As a developer though I would expect something named `resetApplicationStatePromise` to be a promise object.
> 
> I _think_ that just fixing the existing usage of this function will make the problem go away? People copy existing test specs to create new ones, so they will copy the correct usage.
> 
> Also, if we could fix that x-ed out test that would be awesome.

Okay. Reverted the naming.

**Comment by davetsay** [comment]
@akhenry addressed your comments. Can you re-review?

**Comment by akhenry** [comment]
@davetsay Merge conflict 😦

**Comment by nikhilmandlik** [comment]
> @davetsay Merge conflict 😦

this could be my fault, back button fix also uses some of same changes.

**Comment by akhenry** [comment]
@davetsay  @nikhilmandlik  Can we resolve?

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y++
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Review by akhenry — APPROVED** [review]

**Pull request merged**